src/addrman.h

```diff
@@ -468,7 +468,12 @@ class CAddrMan
             LogPrint(BCLog::ADDRMAN, "addrman lost %i new and %i tried addresses due to collisions or invalid addresses\n", nLostUnk, nLost);
         }
 
-        Check();
+        const int check_code{ForcedCheck()};
+        if (check_code != 0) {
+            throw std::ios_base::failure(strprintf(
+                "Corrupt data. Consistency check failed with code %s.",
+                check_code));
```

Comment:
nit: maybe mention "addrman" somewhere in this message. The following looks too generic:
```
2021-08-18T07:12:57Z ERROR: DeserializeDB: Deserialize or I/O error - Corrupt data. Consistency check failed with code -16.: iostream error
```

nit: remove the dot after %s: `with code -16.: iostream error`

Reply:
The new failure mentions the file name `peers.dat`. Is this enough?

```
Error: Invalid or corrupt peers.dat (Corrupt data. Consistency check failed with code -16.: iostream error). If you believe this is a bug, please report it to https://github.com/bitcoin/bitcoin/issues. As a workaround, you can move the file ("/tmp/bitcoin_func_test_ybw3djdd/node0/regtest/peers.dat") out of the way (rename, move, or delete) to have a new one created on the next start.

Reply:
Thanks, removed dot.

Reply:
`peers.dat` looks even better than `addrman`.